.circleci/config.yml

```diff
@@ -1,14 +1,111 @@
+# A note on performance.
+#
+# TL;DR: Storing and retrieving from workspaces is slow so use it sparingly.
+#
+# Workspaces are useful in sharing data between jobs where the overhead
+# involved in using a cache and providing functionality to rebuild the cache in
+# each job is too high.
+#
+# Long version:
+#
+# Storing and retrieving data from workspaces seems to be slow. And
+# caches/checkouts/plain downloads appear to be faster. Some numbers ...
+#
+####
+#
+# Restoring repo dir (no binaries) (65M) + gocache (243M) + gomodcache (1.5G)
+# from a workspace takes approx 121s
+#
+# Restoring repo dir (no binaries) (65M) + gocache (243M) from a workspace
+# takes approx 52s and then restoring the gomodcache (1.5G) from a cache takes
+# approx 12s, so 64s total, so a reduction of 57s which is almost half the
+# time!
+#
+####
+#
+# Restoring monorepo dir + celo-blockchain  + go install (684M) from workspace
+# approx 51s.
+#
+# Restoring monorepo dir + go install (659M) from workspace approx 37s +
+# shallow ceckout celo-monorepo 2s, so a total of 39s a 12 second improvement.
```

Comment:
type "ceckout"

Reply:
Why are both "Restoring monorepo dir" and "shallow "checkout celo-monorepo" needed?